controller/server/main.go

```diff
@@ -14,24 +14,101 @@
 package main
 
 import (
+	"context"
 	"flag"
 	"fmt"
-	"log"
 	"net"
+	"path/filepath"
 
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/util/homedir"
+
+	"github.com/google/kne/deploy"
 	cpb "github.com/google/kne/proto/controller"
-	"google.golang.org/grpc"
+
 	"google.golang.org/grpc/credentials/alts"
+
+	"google.golang.org/grpc"
+)
+
+const (
+	defaultMetallbManifestDir = "../../manifests/metallb"
+	defaultMeshnetManifestDir = "../../manifests/meshnet/base"
 )
 
 var (
-	port = flag.Int("port", 50051, "Controller server port")
+	port           = flag.Int("port", 50051, "Controller server port")
+	defaultKubeCfg = ""
 )
 
 type server struct {
 	cpb.UnimplementedTopologyManagerServer
 }
 
+func getDeployment(req *cpb.CreateClusterRequest) (*deploy.Deployment, error) {
+	d := &deploy.Deployment{}
+	switch kind := req.ClusterSpec.(type) {
+	case *cpb.CreateClusterRequest_Kind:
+		k := &deploy.KindSpec{}
```

Comment:
simpler to write:

d.Cluster = &deploy.KindSpec{
Name: req.GetKind().Name,
...
}

Also, I think there are fields missing here (specifically the GoogleArtifactRegistries field at least)

Reply:
Done and added missing fields